src/ray/common/test/memory_monitor_test.cc

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/common/memory_monitor.h"
+#include "gtest/gtest.h"
+
+namespace ray {
+class MemoryMonitorTest : public ::testing::Test {};
+
+TEST_F(MemoryMonitorTest, TestThresholdZeroAlwaysAboveThreshold) {
+  MemoryMonitor monitor(0 /*usage_threshold*/, 0 /*refresh_interval_ms*/,
+      [](bool is_usage_above_threshold) {
+    FAIL() << "Expected monitor not running";
+  });
+  ASSERT_TRUE(monitor.IsUsageAboveThreshold());
+}
+
+TEST_F(MemoryMonitorTest, TestThresholdOneAlwaysBelowThreshold) {
+  MemoryMonitor monitor(1 /*usage_threshold*/, 0 /*refresh_interval_ms*/,
+      [](bool is_usage_above_threshold) {
+    FAIL() << "Expected monitor not running";
+  });
+  ASSERT_FALSE(monitor.IsUsageAboveThreshold());
+}
+
+TEST_F(MemoryMonitorTest, TestGetNodeAvailableMemoryBytesAlwaysPositive) {
+  {
+    MemoryMonitor monitor(0 /*usage_threshold*/, 0 /*refresh_interval_ms*/,
+        [](bool is_usage_above_threshold) {
+      FAIL() << "Expected monitor not running";
+    });
+    auto [available, total] = monitor.GetNodeAvailableMemoryBytes();
```

Comment:
How do we verify the returend memory usage is actually correct?

Reply:
it is reading the authoritative source, any suggestion on how else we want to verify this?

Reply:
added check against "free"

Reply:
sgtm. We can also probably do some e2e tests later